File: src/main/java/com/gmail/borlandlp/minigamesdtools/util/geometry/GeneratedBlockSchema.java
package com.gmail.borlandlp.minigamesdtools.util.geometry;

import org.bukkit.Location;

import java.util.ArrayList;
import java.util.List;

public class GeneratedBlockSchema {
    private List<Location> borderBlocks;
    private List<Location> fillerBlocks;

    public GeneratedBlockSchema(List<Location> borderBlocks, List<Location> fillerBlocks) {
        this.borderBlocks = borderBlocks;
        this.fillerBlocks = fillerBlocks;
    }

    public List<Location> getBorderBlocks() {
        return this.borderBlocks;
    }

    public List<Location> getFillerBlocks() {
        return this.fillerBlocks;
    }

    public List<Location> getAllBlocks() {
        List<Location> copyBound = new ArrayList<>(this.borderBlocks);//clone Dolly!
        copyBound.addAll(this.fillerBlocks);
        return copyBound;
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/util/geometry/GeneratedBlockSchema.kt
package com.gmail.borlandlp.minigamesdtools.util.geometry

import org.bukkit.Location
import java.util.*

class GeneratedBlockSchema(
    val borderBlocks: MutableList<Location>,
    val fillerBlocks: MutableList<Location>
) {

    val allBlocks: MutableList<Location>
        get() {
            return mutableListOf<Location>().apply {
                this.addAll(fillerBlocks)
                this.addAll(borderBlocks)
            }
        }
}